### PR TITLE
Update to Apollo Client 0.5

### PR DESCRIPTION
**Pull request by stubailo**
Blockers:

- One mutation test failure from https://github.com/apollostack/apollo-client/pull/801

Tests passing:

- [ ] test
- [x] ├── parser.test.ts
- [x] ├── react-native
- [x] │   ├── __snapshots__
- [x] │   │   └── component.test.js.snap
- [x] │   └── component.test.js
- [x] └── react-web
- [ ]     ├── client
- [x]     │   ├── ApolloProvider.test.tsx
- [x]     │   ├── graphql
- [x]     │   │   ├── fragments.test.tsx
- [ ]     │   │   ├── mutations.test.tsx
- [x]     │   │   ├── queries-1.test.tsx
- [x]     │   │   ├── queries-2.test.tsx
- [x]     │   │   ├── shared-operations-1.test.tsx
- [x]     │   │   ├── shared-operations-2.test.tsx
- [x]     │   │   ├── statics.test.tsx
- [x]     │   │   └── subscriptions.test.tsx
- [x]     │   └── libraries
- [x]     │       ├── mobx.test.tsx
- [x]     │       ├── redux-1.test.tsx
- [x]     │       └── redux-2.test.tsx
- [x]     └── server
- [x]         └── index.test.tsx

TODO:

- [ ] If this PR is a new feature, reference an issue where a consensus about the design was reached (not necessary for small changes)
- [ ] Make sure all of the significant new logic is covered by tests
- [ ] Rebase your changes on master so that they can be merged easily
- [ ] Make sure all tests and linter rules pass
- [ ] Update CHANGELOG.md with your change
- [ ] If this was a change that affects the external API, update the docs and post a link to the PR in the discussion

### CODE REVIEW

File: test/react-native/component.test.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ describe('App', () => {
     const query = gql`query people { allPeople(first: 1) { people { name } } }`;
     const data = { allPeople: { people: { name: 'Luke Skywalker' } } };
     const networkInterface = mockNetworkInterface({ request: { query }, result: { data } });
-    const client = new ApolloClient({ networkInterface });
+    const client = new ApolloClient({ networkInterface, addTypename: false });
```
Comment (tmeasday) [inline test/react-native/component.test.js]: Why do this all over the place and default the network interface to supporting it?

Comment (stubailo) [inline test/react-native/component.test.js]: What's the alternative?

Comment (tmeasday) [inline test/react-native/component.test.js]: I thought we discussed making `mockNetworkInterface` support adding `__typename` to results automatically so we didn't have to worry about it in tests.

Comment (tmeasday) [inline test/react-native/component.test.js]: Yeah I guess it just seems like the sort of change that is hard to back out later. But i'll merge it and do that later today.

File: src/graphql.tsx [hunk 2]
```diff
@@ -66,8 +65,18 @@ const defaultMapResultToProps = props => props;
 const defaultMapPropsToSkip = props => false;
 
 // the fields we want to copy over to our data prop
-const observableQueryFields = observable => pick(observable, 'variables',
-  'refetch', 'fetchMore', 'updateQuery', 'startPolling', 'stopPolling');
+function observableQueryFields(observable) {
```
Comment (stubailo) [inline src/graphql.tsx]: @tmeasday @helfer I just realized I forgot to add subscribeToMore here.

Comment (helfer) [inline src/graphql.tsx]: patch release?

Comment (helfer) [inline src/graphql.tsx]: Or do we just wait?

Comment (stubailo) [inline src/graphql.tsx]: I can patch. I don't think there's a big cost to more releases right.

**Pull request merged**